File: python/2108.py
from collections import Counter
import sys
n = int(input())
arr = []
for _ in range(n):
    arr.append(int(input()))
arr.sort()
print("*******************")
print(arr)
print(round(sum(arr) / n))
print(arr[int(n / 2)])
ct = Counter(arr)
if n >= 2 and ct.most_common(2)[0][1] == ct.most_common(2)[1][1]:
    print(ct.most_common(2)[1][0])
else:
    print(ct.most_common(1)[0][0])
print(abs(max(arr) - min(arr)))

File: 9999/2108.cpp
#include<iostream>
#include<algorithm>
#include<vector>
#include<cmath>

using namespace std;

int cnt[8001];
int sum = 0;

int main()
{
    int N;
	cin >> N;

    vector<int> arr;
	for (int i = 0; i < N; i++)
	{
        int k;
		cin >> k;
		sum += k;
		cnt[k+4000]++;
        arr.push_back(k);
	}
	
	sort(arr.begin(), arr.end());
	
	int common;
	int max = 0;
	
	for (int i = 0; i < 8001; i++)
	{
		if (cnt[i] > max)
		{
			max = cnt[i];
			common = i;
		}
	}
    for (int i = common + 1; i < 8001; i++)
	{
		if (cnt[i] == max)
		{
			common = i;
			break;
		}
	}

	cout << (int)round(double(sum) / N) << "\n";
	cout << arr[(N - 1) / 2] << "\n";
	cout << common - 4000 << "\n";
	cout << arr[N - 1] - arr[0] << "\n";
}